### ISSUE TITLE
Moar github compatibility: Header Anchors

### ISSUE THREAD

**Issue by mmterpstra** [joemccann/dillinger#552]
### Question

Can you add header anchors to markdown-it? Because of the github support if it would be nice if you also do it.

### Possible solution
This is on of the markdown-it plugins that will allow to add heading anchors to the headers and thus solve the problem.
https://www.npmjs.com/package/markdown-it-headinganchor

What you can do with the new support:

```md
### TOC

1. [TOC](#toc)
1. [Question](#question)
1. [Possible solution](#possible-solution)
```

Renders to ( works better with longer pages, so i inserted some useless `<br>` at the end): 

### TOC

1. [TOC](#toc)
1. [Question](#question)
1. [Possible solution](#possible-solution)

<br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br>

**Comment by joemccann** [comment]
appears this needs to just add the markdown-it plugin to the build/bundle?

**Comment by mmterpstra** [comment]
Yes, but a few thing need to be tested.
 - Working achors
 - The achor links work in the preview and do not mess up scrolling
 - Check for reserved things on http://dillinger.io/# ...
 - maybe insert an example in the example doc and test it.(see below)

```
### Table of contents

1. [Table of contents](#table-of-contents)
2. [Dillinger](#dillinger)
   1.  [Version](#version)
   2. [Tech](#tech)
   3. [Installation](#installation)
   4. [Plugins](#plugins)
   5. [Development](#development)
   6. [Docker](#docker)
   7. [N|Solid and NGINX](#n-solid-and-nginx)
   8. [docker-compose.yml](#docker-compose.yml)
   9. [Todos](#todos)
```

**Comment by joemccann** [comment]
This seems to work fine on Dillinger.io?

**Comment by mmterpstra** [comment]
I did not test it because I did not have time/experience to set this up/test this I hoped the community would pick this up.

If you like it you could add this to the site.

**Comment by joemccann** [comment]
yeah this _doesn't_ work currently but I'm sure there's some JavaScript wizardry to fix this.

**Comment by sneakernets** [comment]
I ran into this when I used RedCarpet, the Ruby Markdown library. Wasn't so much header anchors, just anchors in general didn't work.

This is essentially what I came up with:

```ruby
    def link(url, title, text)
      if url.start_with?("id:")
        %(<span id="#{url.split(":").last}">#{text}</span>)
      else
        %(<a href="#{url}" title="#{title}">#{text}</a>)
      end
    end
```

So with this, I could write:
```
(in table of contents)
### [Getting Started](#getting_started)

(in actual document)
## [Getting Started](id:getting_started)
```
And be able to click the first link to drop down to that header in the document.

I realize that's ruby, but it's *something*. 😄 

NOTE: I've been told that in the HTML5 spec, using "id=" like I did can actually break javascript (somehow), and to use "name=" instead. Thing is, that's been *deprecated* in XHTML.
